### ISSUE TITLE
Better error message when SSH host key validation fails

### ISSUE THREAD

**Issue by heapcrash** [Gallopsled/pwntools#1749]
When connecting via `ssh()` to a remote host, sometimes the host key changes (especially with hosted wargames like `leviathan.labs.overthewire.org`).

When this fails, the following exception is thrown:

```
>>> from pwn import *
>>> ssh(user='leviathan0', password='leviathan0', host='leviathan.labs.overthewire.org', port=2223)
[x] Connecting to leviathan.labs.overthewire.org on port 2223
[-] Connecting to leviathan.labs.overthewire.org on port 2223: Failed
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/heapcrash/pwntools/pwnlib/tubes/ssh.py", line 664, in __init__
    self.client.connect(host, port, user, password, key, keyfiles, self.timeout, compress = True)
  File "/Users/heapcrash/.pyenv/versions/3.8.3/lib/python3.8/site-packages/paramiko/client.py", line 423, in connect
    raise BadHostKeyException(hostname, server_key, our_key)
paramiko.ssh_exception.BadHostKeyException: Host key for server 'leviathan.labs.overthewire.org' does not match: got 'AAAAC3NzaC1lZDI1NTE5AAAAIIJwHwzTXHJomqOQfM8emYvBLMkHQrtG5nBz6MP/No4O', expected 'AAAAC3NzaC1lZDI1NTE5AAAAIDPNAn6LOV1QtSqlnnXz6eECuznq5bNIe6DLFpMNZMEI'
```

We should catch this exception and suggest to the user a way to circumvent this?  I'm not aware of any parameters to the `ssh()` tube that ignore host keys.  Specifically, we ask Paramiko to enforce this policy:

https://github.com/Gallopsled/pwntools/blob/9de743810efc9ed9520bf06f7544c4db8f46058c/pwnlib/tubes/ssh.py#L644-L646

Since this is a non-default setting that Pwntools enforce, it seems that we should catch this exception and AT LEAST add an additional warning message suggesting how it might be fixed.  The easiest way that I can think of is:

```
$ ssh-keygen -R "leviathan.labs.overthewire.org"
```

What do you think, @Arusekk?

**Comment by heapcrash** [comment]
Apparently the `ssh-keygen -R` command is insufficient.

```sh
$ ssh-keygen -R 'leviathan.labs.overthewire.org'
Host leviathan.labs.overthewire.org not found in /Users/heapcrash/.ssh/known_hosts
$ grep leviathan ~/.ssh/known_hosts
[leviathan.labs.overthewire.org]:2223 ssh-ed25519 AAAAC3NzaC1lZDI1NTE5AAAAIDPNAn6LOV1QtSqlnnXz6eECuznq5bNIe6DLFpMNZMEI
```

**Comment by Arusekk** [comment]
I'm totally in favor of gracefully handling all problems not caused by exploit code or things in pwntools. We could probably lift openssh warning template:
```
@ @ @ @ @ @ @ @ @ @ @ @ @ @ @ @ @ @ @ @ @ @ @ @ @ @ @ @ @ @
@    WARNING: REMOTE HOST IDENTIFICATION HAS CHANGED!     @
@ @ @ @ @ @ @ @ @ @ @ @ @ @ @ @ @ @ @ @ @ @ @ @ @ @ @ @ @ @
IT IS POSSIBLE THAT SOMEONE IS DOING SOMETHING NASTY!
Someone could be eavesdropping on you right now (man-in-the-middle attack)!
It is also possible that a host key has just been changed.
The fingerprint for the ED25519 key sent by the remote host is
SHA256:AAAAC3NzaC1lZDI1NTE5AAAAIDPNAn6LOV1QtSqlnnXz6eECuznq5bNIe6DLFpMNZMEI.
Please contact your system administrator.
Add correct host key in /home/heapcrash/.ssh/known_hosts to get rid of this message.
Offending ED25519 key in /home/heapcrash/.ssh/known_hosts:11
  remove with:
  ssh-keygen -f "/home/heapcrash/.ssh/known_hosts" -R "[leviathan.labs.overthewire.org]:2223"
ED25519 host key for leviathan.labs.overthewire.org has changed and you have requested strict checking.
Host key verification failed.
```

**Comment by zachriggle** [comment]
It looks like you also need to specify the port when calling `ssh-keygen -R`.  See this StackOverflow post here:
https://superuser.com/questions/1611832/why-doesnt-ssh-keygen-r-remove-the-host-from-ssh-known-hosts/1611834#1611834

**Comment by Arusekk** [comment]
Closed by #1770
